catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -538,4 +642,72 @@ private void submitToCatalog(ScheduledExecutorService batchScheduler,
             }
         }, 100, 100, TimeUnit.MILLISECONDS);
     }
+
+    private int getFilesCount(File rootFile) {
+        if (!rootFile.isDirectory()) {
+            return 1;
+        }
+
+        File[] files = rootFile.listFiles();
+        int count = 0;
+        for (File file : files) {
+            if (file.isDirectory()) {
+                count += getFilesCount(file);
+            } else {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    private Map<String, List<File>> generateFileMap() {
+
+        File inputFile = new File(filePath);
+        File[] files;
+        if (!inputFile.isDirectory()) {
+            File contentDirectory = new File(inputFile.getParent() + CONTENT_PATH);
+            files = contentDirectory.listFiles();
```

Comment:
Same thing here about `listFiles()` being slow. Try to avoid doing this and simply walk the file tree instead, doing whatever you need to do along the way.